scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+ cat << EOF > "$CLONE_SQL"
+ SET SESSION sql_log_bin=OFF;
+ CREATE USER '$CLONE_USER'@'%' IDENTIFIED $WITH_OPTION BY '$CLONE_PSWD';
+ GRANT CLONE_ADMIN, SYSTEM_VARIABLES_ADMIN, SUPER, SHUTDOWN, EXECUTE ON *.* to '$CLONE_USER'@'%';
+ GRANT INSERT ON mysql.plugin to '$CLONE_USER';
+ GRANT SELECT,UPDATE,INSERT ON performance_schema.* TO '$CLONE_USER';
+ CREATE USER '$CLONE_USER'@'localhost' IDENTIFIED $WITH_OPTION BY '$CLONE_PSWD';
+ GRANT CLONE_ADMIN, SYSTEM_VARIABLES_ADMIN, SUPER, SHUTDOWN, EXECUTE ON *.* TO '$CLONE_USER'@'localhost';
+ GRANT EXECUTE ON *.* to '$CLONE_USER'@'localhost';
+ GRANT INSERT ON mysql.plugin to '$CLONE_USER'@'localhost';
+ GRANT SELECT,UPDATE,INSERT ON performance_schema.* TO '$CLONE_USER'@'localhost';
+EOF
+
+    wsrep_log_info "Launching clone recipient daemon"
+    wsrep_log_info "-> using: $CLONE_ENV $CLONE_BINARY_SAFE $DEFAULT_OPTIONS "
+    wsrep_log_debug "-> Test connection as: -u$CLONE_USER -pxxxxxx -h $JOINER_CLONE_HOST -P $JOINER_CLONE_PORT"
+
+    # Define client to be used on the Joiner side
+    MYSQL_ACLIENT="$MYSQL_CLIENT -u$CLONE_USER -S$CLONE_SOCK --batch --skip_column_names --silent"
+    wsrep_log_debug "-> MYSQL_ACLIENT: $MYSQL_ACLIENT"
+
+    # HERE We start the instance that will get the clone and pass the CLONE_SQL file to create user
+    eval WSREP_SST_CLONE=1 $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS \
+        --wsrep_provider=none --init_file="$CLONE_SQL" >> $CLONE_ERR &
+
+    wsrep_log_info "-> Initialization user returned $?"
+
+    if [ ! "$?" == "0" ]; then
+        wsrep_log_error "-> User Creation on Joiner node failed, possible permission denied. Check permissions for "
+        #we will try to silently shutdown the instance
+        `$MYSQL_ACLIENT -e "SHUTDOWN" 2> /dev/null` || :
+        exit 1
+    fi
+
+    # Wait for the receiver process to start
+    until [ -n "$(cat $CLONE_PID_FILE 2>/dev/null)" ]
+    do
+        sleep 0.2
+    done
+    CLONE_INSTANCE_PID=`cat $CLONE_PID_FILE`
+
+    if [ -n "$WSREP_LOG_DEBUG" ]; then
+        GRANTS=`$MYSQL_ACLIENT -uroot -NB -e "show grants for '$CLONE_USER'@'localhost'"`
+        wsrep_log_debug "-> Clone user grants: $GRANTS"
+    fi
+
+    export MYSQL_PWD=$CLONE_PSWD
+#    wsrep_log_debug "-> Exported MySQL password $MYSQL_PWD"
+
+    wsrep_log_info "Waiting for clone recipient daemon to be ready for connections at port $JOINER_CLONE_PORT"
+    wsrep_log_debug "-> connecting as: $MYSQL_CLIENT -u$CLONE_USER -h$JOINER_CLONE_HOST -P$JOINER_CLONE_PORT"
+    to_wait=$JOINER_TIMEOUT_WAIT_CLONE_INSTANCE # 90 sec by defaults but it can be changed in the config
+    until $MYSQL_CLIENT -u$CLONE_USER -h$JOINER_CLONE_HOST -P$JOINER_CLONE_PORT \
+          -e 'SELECT USER()' > /dev/null
+    do
+        if [ $to_wait -eq 0 ]
+        then
+            wsrep_log_error "Timeout waiting for clone recipient daemon"
+            exit $ETIMEDOUT
+        fi
+        to_wait=$(( $to_wait - 1 ))
```

Comment:
SC2004: $/${} is unnecessary on arithmetic variables.

Reply:
fixed